pkg/cloud/azure/services/groups/groups.go

```diff
@@ -23,28 +23,79 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/pkg/errors"
 	"k8s.io/klog"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1alpha1"
 	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/converters"
 )
 
+// Reconcile reconciles information about a resource.
+func (s *Service) Reconcile(ctx context.Context, _ azure.Spec) error {
+	klog.V(2).Infof("reconciling resource group %s", s.Scope.ResourceGroup().Name)
+	rg, err := s.Get(s.Scope.Context)
+	if err != nil {
+		switch {
+		case s.Scope.ResourceGroup().IsProvided():
```

Comment:
Create a Spec which specifies the name and Get/CreateOrUpdate/Delete just works on that spec rather than polluting this function.

That way a reconciler does a Get -> if managed does not call createorupdate, similar to reconciler delete -> does a get if managed does not call service.delete.

Reply:
See my comments below and #110.